plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/CostingFun/Core.hs

```diff
@@ -179,6 +180,16 @@ scaleLinearly (Intercept intercept) (Slope slope) =
     addCostStream (CostLast intercept) . mapCostStream (slope *)
 {-# INLINE scaleLinearly #-}
 
+-- | Take an intercept, two slopes and two streams, and scale each element of
+-- the first stream by the first slope, each element of the second stream by the
+-- second slope, add the two scaled streams together, and cons the intercept to
+-- the stream afterwards.
+scaleLinearlyTwoVariables :: Intercept -> Slope -> CostStream -> Slope -> CostStream -> CostStream
+scaleLinearlyTwoVariables (Intercept intercept) (Slope slope1) stream1 (Slope slope2) stream2 =
+    addCostStream (CostLast intercept) $
+                  addCostStream (mapCostStream (slope1 *) stream1) (mapCostStream (slope2 *) stream2)
```

Comment:
Nit: weird indentation. Otherwise looks good to me.

Reply:
Should be fixed now.